test/functional/rpc_blockchain.py

```diff
@@ -388,6 +388,17 @@ def _test_gettxoutsetinfo(self):
         # Unknown hash_type raises an error
         assert_raises_rpc_error(-8, "'foo hash' is not a valid hash_type", node.gettxoutsetinfo, "foo hash")
 
+        self.restart_node(
+            0,
+            extra_args=[
+                "-coinstatsindex=1",
+                "-reindex"
+            ],
+        )
+        # Block not found error
+        assert_raises_rpc_error(-5, "Block not found", self.nodes[0].gettxoutsetinfo, "none", "00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09", True)
```

Comment:
Looks like this is checking for a block hash that doesn't exist after block hash for block 1 is reconsidered, which makes sense.  Recommend adding `self.log.info()` to make this clearer to the reader.

```python
b1hash = node.getblockhash(1)
node.invalidateblock(b1hash)
...
self.log.info("Test gettxoutsetinfo returns the same result after invalidate/reconsider block")
node.reconsiderblock(b1hash)
```

Reply:
do you think I should still add this after f169562d85b332576b3a1cf32755e43023227ed9, rather than invalidating a block to then use `gettxoutsetinfo` with its hash I just used a random block hash. I can try doing this aswell

Reply:
Looks like that's the block hash for block 1000 (mainnet).  What's the regtest minimum difficulty?  The hash that we don't want to be found should be something that definitely isn't possible.

Would also be good to inform the log, for example:
```diff
diff --git a/test/functional/feature_coinstatsindex.py b/test/functional/feature_coinstatsindex.py
index c6edbe7c31..8e37a3c537 100755
--- a/test/functional/feature_coinstatsindex.py
+++ b/test/functional/feature_coinstatsindex.py
@@ -242,6 +242,7 @@ class CoinStatsIndexTest(BitcoinTestFramework):
         res12 = index_node.gettxoutsetinfo('muhash')
         assert_equal(res12, res10)
 
+        self.log.info("Test obtaining info for a non-existent block hash")
         assert_raises_rpc_error(-5, "Block not found", index_node.gettxoutsetinfo, hash_type="none", hash_or_height="00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09", use_index=True)
 
     def _test_use_index_option(self):
```

Reply:
thanks updated in [6809ffd](https://github.com/bitcoin/bitcoin/pull/30340/commits/6809ffd8a6c589c515af48da2dd8367e4c6c2c26)

also the earlier comment I addressed below https://github.com/bitcoin/bitcoin/pull/30340#issuecomment-2195027831